drivers/max31865/max31865.c

```diff
@@ -0,0 +1,334 @@
+
+/**
+ * @brief Convert raw data to temperature
+ * @param[in] dev Device descriptor of the driver
+ * @param[in] raw_data Code used to set the RTD low and RTD high registers
+ * @param[out] temp Temperature in Celsius centi-degrees (0.01°C)
+ */
+int _raw_to_temperature(max31865_t *dev, uint16_t raw_data, int32_t *temperature)
+{
+    assert(dev->params->lut);
+    assert(temperature);
+
+    if(raw_data < (*dev->params->lut)[0][MAX31865_LUTCOL_CODE]) {
+        LOG_ERROR("%s() >> ERROR: raw_data too small 0x%04X < 0x%04X\n", __FUNCTION__, raw_data, (uint16_t)(*dev->params->lut)[0][MAX31865_LUTCOL_CODE]);
+        return -EINVAL;
+    }
+    if(raw_data > (*dev->params->lut)[dev->params->lut_numlines-1][MAX31865_LUTCOL_CODE]) {
+        LOG_ERROR("%s() >> ERROR: raw_data too big 0x%04X > 0x%04X\n", __FUNCTION__, raw_data, (uint16_t)(*dev->params->lut)[dev->params->lut_numlines-1][MAX31865_LUTCOL_CODE]);
+        return -EINVAL;
+    }
+    /* walk the LUT to find the appropriate coefficients for linear interpolation: */
+    int i = 0;
+    for(i=0 ; i<dev->params->lut_numlines ; i++) {
+        if(raw_data < (*dev->params->lut)[i][MAX31865_LUTCOL_CODE])
+            break;
+    }
+    i--;
+    LOG_DEBUG("%s() >> i = %d\n", __FUNCTION__, i);
+    LOG_DEBUG("%s() >> a0 = %ld, a1 = %ld, raw_data = 0x%04X\n", __FUNCTION__,
+           (*dev->params->lut)[i][MAX31865_LUTCOL_A0], (*dev->params->lut)[i][MAX31865_LUTCOL_A1],
+           (uint16_t)(*dev->params->lut)[i][MAX31865_LUTCOL_CODE]);
+    /* calculate T in µ°C by linear interpolation with the coefficients from the LUT: */
+    int32_t temp_uc = (*dev->params->lut)[i][MAX31865_LUTCOL_A0]
+        + (*dev->params->lut)[i][MAX31865_LUTCOL_A1] * raw_data;
+    /* convert µ°C to c°C: */
+    int32_t temp_cc = temp_uc / 10000;
+    LOG_DEBUG("%s() >> T (°µC) = %ld\n", __FUNCTION__, temp_uc);
+    LOG_DEBUG("%s() >> T (°C) = %ld.%02d\n", __FUNCTION__, temp_cc / 100, abs(temp_cc) % 100);
+
+    /* convert to centi degC */
+    *temperature = temp_uc / 10000;
+
+    return 0;
+}
+
+/**
+ * @brief Convert temperature to raw data
+ * @param[in] dev Device descriptor of the driver
+ * @param[in] temp Temperature in Celsius centi-degrees (0.01°C)
+ * @param[out] raw_data Code used to set the RTD low and RTD high registers
+ * @return 0 on success.
+ * @return -EINVAL if temperature is out of LUT range.
+ */
+int _temp_to_raw(max31865_t *dev, int32_t temp, uint16_t *raw_data)
+{
+    int32_t temp_uc = temp * 10000;     // c°C --> µ°C
+
+    assert(raw_data);
+    assert(dev->params->lut_numlines);
+    if( (temp_uc < (*dev->params->lut)[0][MAX31865_LUTCOL_TEMP])
+        || (temp_uc > (*dev->params->lut)[dev->params->lut_numlines - 1][MAX31865_LUTCOL_TEMP]) )
+        return -EINVAL;
+
+    int i = 0;
+    for(i=0 ; i<dev->params->lut_numlines ; i++) {
+        if(temp_uc < (*dev->params->lut)[i][MAX31865_LUTCOL_TEMP])
+            break;
```

Comment:
```suggestion
        if (temp_uc < (*dev->params->lut)[i][MAX31865_LUTCOL_TEMP]) {
            break;
        }
```

This is just an example, but our Coding Convention says that even single line if-statements should have curly braces and the `if` should be followed by a whitespace: https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md#indentation-and-braces